File: tests/code/corpus/hard/php/Advanced.php
<?php

namespace App\Advanced;

enum Status: string
{
    case Active = 'active';
    case Frozen = 'frozen';

    public function label(): string
    {
        return $this->value;
    }
}

abstract class Base
{
    abstract public function render(): string;

    public static function create(): static
    {
        return new static();
    }
}

final class Impl extends Base
{
    public function render(): string
    {
        return $this->helper();
    }

    private function helper(): string
    {
        return 'x';
    }
}

$anonymous = new class extends Base {
    public function render(): string
    {
        return 'anon';
    }
};

$arrow = fn ($x) => $x * 2;

File: tests/code/corpus/hard/swift/Advanced.swift
import Foundation

enum Status {
    case active
    case frozen

    func label() -> String {
        return "status"
    }
}

extension String {
    func shout() -> String {
        return self.uppercased()
    }
}

struct Holder<T> {
    var value: T

    func unwrap() -> T {
        return value
    }
}

func combine<T>(_ left: T, _ right: T) -> [T] {
    return [left, right]
}

protocol Renderable {
    associatedtype Output
    func render() -> Output
}
